src/hooks/useUpdateCart.ts

```diff
@@ -0,0 +1,22 @@
+import { useContext, useState } from 'react';
+import DataContext from '../context/data.context';
+
+function useUpdateCart() {
+  const ctx = useContext(DataContext);
+  const storageCart = JSON.parse(localStorage.getItem('cart') || '[]');
+  Object.assign(ctx.cart.idArray, storageCart);
```

Comment:
It looks not so good. Changing the properties of a context object directly is not the best practice.